ui/src/clockface/components/resource_list/ResourceEditableName.tsx

```diff
@@ -0,0 +1,172 @@
+// Libraries
+import React, {Component, KeyboardEvent, ChangeEvent, MouseEvent} from 'react'
+import classnames from 'classnames'
+
+// Components
+import {Input, SpinnerContainer, TechnoSpinner} from '@influxdata/clockface'
+import {ClickOutside} from 'src/shared/components/ClickOutside'
+
+// Types
+import {ComponentSize} from '@influxdata/clockface'
+import {RemoteDataState} from 'src/types'
+
+// Decorators
+import {ErrorHandling} from 'src/shared/decorators/errors'
+
+interface Props {
+  onUpdate: (name: string) => void
+  name: string
+  onClick?: (e: MouseEvent<HTMLAnchorElement>) => void
+  placeholder?: string
+  noNameString: string
+  parentTestID: string
+  buttonTestID: string
+  inputTestID: string
+  hrefValue: string
+}
+
+interface State {
+  isEditing: boolean
+  workingName: string
+  loading: RemoteDataState
+}
+
+@ErrorHandling
+class ResourceEditableName extends Component<Props, State> {
+  public static defaultProps = {
+    parentTestID: 'resource-editable-name',
+    buttonTestID: 'resource-editable-name--button',
+    inputTestID: 'resource-editable-name--input',
+    hrefValue: '#',
+  }
+
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      isEditing: false,
+      workingName: props.name,
+      loading: RemoteDataState.Done,
+    }
+  }
+
+  public render() {
+    const {
+      name,
+      hrefValue,
+      noNameString,
+      parentTestID,
+      buttonTestID,
+    } = this.props
+
+    return (
+      <div className={this.className} data-testid={parentTestID}>
+        <SpinnerContainer
+          loading={this.state.loading}
+          spinnerComponent={<TechnoSpinner diameterPixels={20} />}
+        >
+          <a href={hrefValue} onClick={this.handleClick}>
+            <span>{name || noNameString}</span>
+          </a>
+        </SpinnerContainer>
+        <div
+          className="resource-editable-name--toggle"
+          onClick={this.handleStartEditing}
+          data-testid={buttonTestID}
+        >
+          <span className="icon pencil" />
+        </div>
+        {this.input}
+      </div>
+    )
+  }
+
+  private get input(): JSX.Element {
+    const {placeholder, inputTestID} = this.props
```

Comment:
Could we extract this to its own component? Usually a getter for a JSX element is usually a good indicator.